modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/api/SessionBuilderImpl.java

```diff
@@ -103,20 +106,32 @@ public SessionBuilder defaultPageSize(int pageSize) {
     /** {@inheritDoc} */
     @Override
     public SessionBuilder property(String name, @Nullable Object value) {
-        props.put(name, value);
+        var prop = QueryProperty.byName(name);
```

Comment:
What will be if server supports a property, but client doesn't?

Reply:
This is a sessionBuilder for embedded session, so the set of properties will always be the same.